src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -275,53 +294,105 @@ protected override void ProcessRecord()
         protected override void StopProcessing()
         {
             _sender?.SendAsyncCancel();
+            cancellationTokenSource.Cancel();
         }
 
         #region ConnectionTest
 
+        private void SetCountForTcpTest()
+        {
+            if (Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count = 1;
+            }
+        }
+
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            SetCountForTcpTest();
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
-
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
+                long latency = 0;
+                SocketError status = SocketError.SocketError;
+
+                Stopwatch stopwatch = new Stopwatch();
+
+                using TcpClient client = new TcpClient();
 
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
-                    Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Start();
 
-                    if (timeoutTask.Status == TaskStatus.Faulted || timeoutTask.Status == TaskStatus.Canceled)
+                    if (client.ConnectAsync(targetAddress, TcpPort).Wait(TimeoutSeconds * 1000, cancellationToken))
```

Comment:
ConnectAsync() returns a Task. Please use a pattern from #11517.

Reply:
It is still not addressed.

Reply:
For sure! I was going to jump on that as soon as I can! I made some of the small changes this morning, but am going to dig into things this afternoon 😊

Reply:
I've looked at this a bit, and am not clear on how we'd implement the pattern from #11517. 

I see that `ConnectAsync()` returns a `Task`, compared with `SendPingAsync()` returning `Task<PingReply>` . In the first case, we're using the completion of the task itself as part of the result, compared with the actual result data being returned in the second example. 

I'm not sure how we'd implement this, while also including a way to force the timeout. 

Are you able to provide a bit more detail? I may be interpreting things the wrong way 😁

Thanks!!

Reply:
@jackdcasey  Sorry, I am ok with the change (I skipped Wait() on line end).

Reply:
No problem, I'll get this one resolved 😁